contrib/migration_script.py

```diff
@@ -0,0 +1,170 @@
+    )
+    parser.add_argument(
+        '--server',
+        required=True,
+        # default='http://tronplayground-uswest1adevc.dev.yelpcorp.com:8089',
+        help='specify the location of tron master'
+    )
+    parser.add_argument(
+        '--old-ns',
+        required=True,
+        # default='dsc-test-service',
+        help='Old namespace'
+    )
+    parser.add_argument(
+        '--new-ns',
+        required=True,
+        # default='test_namespace',
+        help='New namespace'
+    )
+    parser.add_argument(
+        'source',
+        help='source file to get list of jobs'
+    )
+    args = parser.parse_args()
+    return args
+
+
+def check_job_if_running(jobs_status, job_name):
+    for job_status in jobs_status:
+        if job_status['name'] == job_name:
+            status = job_status['status']
+            if status == 'running':
+                print(bcolors.FAIL + 'job {} is still running, can not migrate'.format(job_name) + bcolors.ENDC)
+                return False
+            else:
+                print(bcolors.OKGREEN + 'job {} is not running, can migrate'.format(job_name) + bcolors.ENDC)
+                return True
+
+    print(bcolors.FAIL + 'Can not find the job {}'.format(job_name) + bcolors.ENDC)
+    return False
+
+
+def command_jobs(command, jobs, args):
+    data = {'command': command}
+    command_flag = True
+    for job in jobs:
+        if command == 'enable':
+            job_name = args.new_ns + '.' + job['name']
+        else:
+            job_name = args.old_ns + '.' + job['name']
+
+        if command == 'move':
+            data = {'command': command, 'old_name': args.old_ns + '.' + job['name'], 'new_name': args.new_ns + '.' + job['name']}
+            uri = urljoin(args.server, 'api/jobs')
+        else:
+            data = {'command': command}
+            uri = urljoin(args.server, 'api/jobs/' + job_name)
+
+        response = client.request(uri, data=data)
+        if response.error:
+            print(bcolors.FAIL + 'Failed to {} {}'.format(command, job_name) + bcolors.ENDC)
+            command_flag = False
+        else:
+            print(bcolors.OKGREEN + 'Succeed to {} {}'.format(command, job_name) + bcolors.ENDC)
+    return command_flag
+
+
+def ssh_command(hostname, command):
+    ssh = subprocess.Popen(["ssh", "-A", hostname, command], shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    result = ssh.stdout.readlines()
+    error = ssh.stderr.readlines()
+    if error != []:
+        print(bcolors.BOLD + 'Execute command {}: {}'.format(command, error) + bcolors.ENDC)
+    return result
+
+
+def main():
+    args = parse_args()
+    filename = args.source
+    hostname = urlparse(args.server).hostname
+    if filename.endswith(".yaml"):
+        tron_client = client.Client(args.server)
+        jobs_status = tron_client.jobs()
+
+        migration_flag = True
+        with open(filename, "r") as f:
+            jobs = yaml.load(f)['jobs']
```

Comment:
To make this easier to run on partial namespaces (or single jobs), can we extract out the rest of the code into a separate function, that takes a list of jobs?
(could do in a follow-up, if you want to start on full namespaces now)

Reply:
Yeah, will do in a follow-up PR.